cheatsheet.md

```diff
@@ -0,0 +1,162 @@
+
+This can hint at a local `inline` when you see something like `function__##currentNamespace_varation##currentNamespace`, for example
+`getItemNo__15daSpcItem01_prmFP13daSpcItem01_c`. That same format can hint at another file's inline being referenced.
+
+## Macros
+
+The two main Macros you'll likely see are the `fopAcm_SetupActor` Macro, and the `JUTAssertion` Macros.
+
+### fopAcm_SetupActor
+For `fopAcm_SetupActor` the first hint is being in some sort of `create` function, in Ghidra the macro is already expanded
+so just keep an eye out for something like the following.
+```c++
+if (((this->parent).mCondition & Constructed) == 0) {
+    if(this != (structPointer_c *) 0x0) {
+        fopAc_ac_c::fopAc_ac_c((fopAc_ac_c*) this);
+        // More stuff
+    }
+    (this->parent).mCondition = (this->parent).mCondition | Constructed;
+}
+```
+Everything within the first `if` statement is handled by the `fopAcm_SetupActor` Macro.
+
+### JUTAssertion
+This is more a "family" of Macros, however most of the time we're dealing with `JUT_ASSERT`. Super easy pattern to spot,
+just the following.
+```c++
+if (condition) {
+  tempVar = JUTAssertion::getSDevice();
+  JUTAssertion::showAssert(tempVar, "current_file.cpp", lineNumber, "condition");
+  m_Do_printf::OSPanic("current_file.cpp", lineNumber, "Halt");
+}
+```
+This is really useful for figuring out the variable names used, so keep an eye out for it. See `JUTAssert.h` for the
+full family of Macros.
+
+## Recognize the 'switch'
+The original developers of The Wind Waker used switch statements extensively. Ghidra fails to recognize small switches,
+so here is a quick example to refer to.
+
+```c++
+if (m_itemNo != SHIELD) {
+    if (m_itemNo < SHIELD && m_itemNo == PENDANT) {
+        if (mAcch.m_flags & dBgS_Acch::GROUND_LANDING) {
+            // stuff
+        }
+    } else if (mAcch.m_flags & dBgS_Acch::GROUND_LANDING) {
+        // stuff 1
+    }
+}
+// Should be
+switch (m_itemNo) {
+case SHIELD:
+    break;
+case PENDANT:
+    if (mAcch.ChkGroundLanding()) {
+        // stuff
+    }
+    break;
+default:
+    if (mAcch.ChkGroundLanding()) {
+       // stuff 1
+    }
+    break;
+}
+```
+You can also get single case switch statements instead of `if/else`. For example,
+```c++
+if (m_itemNo == BOKO_BELT) {
+    offsetY = -24.0f;
+}
+// Should be
+switch (m_itemNo) {
+case BOKO_BELT:
+    offsetY = -24.0f;
+    break;
+}
+```
+The above is especially useful to try out in case you're dealing with register mismatch.
+The assembly will hint at as the switch statement if you get this pattern,
+```asm
+cmpwi
+beq
+bge
+;
+cmpwi
+beq
+b
+;; Aka cmpwi + a lot of branching is likely a small switch statement.
+```
+
+## Register Problems
+When dealing with register mismatching, we've found the following situations to help.
+When dealing with enums, you might have to do a double return. For example
+```c++
+return createInit() ? 4 : 5 
+// Matches, but we know Enums were used in this function.
```

Comment:
nitpick: technically I don't think we know that 100% for sure (no asserts in this function)